modules/flowable-engine/src/main/java/org/flowable/engine/impl/util/EventSubscriptionUtil.java

```diff
@@ -134,5 +139,16 @@ protected static void scheduleEventAsync(EventSubscriptionEntity eventSubscripti
         }
 
         jobService.scheduleAsyncJob(message);
+
+        if (StringUtils.isEmpty(executionId) && payload != null) {
```

Comment:
@tijsrademakers : The payload is stored in variables only for start events (intermediate events variables are handled in the different way). From my point of view we could use the same approach for the both of them.